core/models/flow_stats.go

```diff
@@ -30,54 +30,47 @@ func (s segmentID) String() string {
 	return fmt.Sprintf("%s:%s", s.exitUUID, s.destUUID)
 }
 
-type segmentContact struct {
+type segmentRecentContact struct {
 	contact *flows.Contact
 	operand string
 	time    time.Time
+	rnd     string
 }
 
 // RecordFlowStatistics records statistics from the given parallel slices of sessions and sprints
 func RecordFlowStatistics(ctx context.Context, rt *runtime.Runtime, tx *sqlx.Tx, sessions []flows.Session, sprints []flows.Sprint) error {
-	rc := rt.RP.Get()
-	defer rc.Close()
-
-	segmentIDs := make([]segmentID, 0, 10)
```

Comment:
pretty sure this was just to let iterate `recentBySegment` in same order we find segments making the allocation of random values deterministic... making the test always pass.. but doing the random value generation when we're going thru segments has the same effect